crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1055,12 +1055,12 @@ impl Backend {
             header.base_fee_per_gas.unwrap_or_default(),
         );
 
-        // notify all listeners
-        self.notify_on_new_block(header, block_hash);
-
         // update next base fee
         self.fees.set_base_fee(next_block_base_fee);
 
+        // notify all listeners
+        self.notify_on_new_block(header, block_hash);
+
```

Comment:
this was a subject to race condition anyway, but this just ensures that value in `fees` is always up to date

Reply:
right, nice catch